py/desispec/sky.py

```diff
@@ -982,25 +982,63 @@ def subtract_sky(frame, skymodel, apply_throughput_correction = False, zero_ivar
-        # need to fit for a multiplicative factor of the sky model
-        # before subtraction
-        # we are going to use a set of bright sky lines,
-        # and fit a multiplicative factor + background around
-        # each of them individually, and then combine the results
-        # with outlier rejection in case a source emission line
-        # coincides with one of the sky lines.
-        for fiber in range(frame.flux.shape[0]) :
-            # apply this correction to the sky model even if we have not fit it (default can be 1 or 0)
-            skymodel.flux[fiber] *= skymodel.throughput_corrections[fiber]
+    if skymodel.throughput_corrections is not None :
+        # a multiplicative factor + background around
+        # each of the bright sky lines has been fit.
+        # here we apply this correction to the emission lines only or to the whole
+        # sky spectrum
+        if apply_throughput_correction  :
+            for fiber in range(frame.flux.shape[0]) :
+                # apply this correction to the sky model even if we have not fit it (default can be 1 or 0)
+                skymodel.flux[fiber] *= skymodel.throughput_corrections[fiber]
+
+        elif apply_throughput_correction_to_lines :
+
+            # we first identify bright lines on the sky model
+
+            # median sky of all fibers
+            msky  = np.median(skymodel.flux,axis=0)
+            # the continuum is the spectrum within 50 e/A of this interpolated minimal values
+            threshold = 50. # e/A
+
+            # find minima in sky spectrum
+            minima = np.zeros(msky.shape)
+            minima[1:-1] = (msky[1:-1]<msky[:-2])&(msky[1:-1]<msky[2:])&(msky[1:-1]>-20)&(msky[1:-1]<0.5*np.max(msky))
+            minima  = np.where(minima)[0]
+            if minima.size > 0 :
+                cont    = np.interp(skymodel.wave,skymodel.wave[minima],msky[minima])
+                # second minima
+                minima2 = np.zeros(msky.shape)
+                minima2[1:-1] = (cont[1:-1]<cont[:-2])&(cont[1:-1]<cont[2:])&(cont[1:-1]>-20)&(cont[1:-1]<0.5*np.max(cont))
+                minima2  = np.where(minima2)[0]
+                cont    = np.interp(skymodel.wave,skymodel.wave[minima2],msky[minima2])
+                in_cont = np.where(np.abs(msky-cont)<threshold)[0] # keep all pixels without n sigma of cont for debiasing estimate
+
+            else :
+                in_cont = np.array([])
+                log.warning("Could not separate the sky continuum from the lines.")
+
+            if in_cont.size > 0 :
+                # apply this correction to the sky lines only
+                for fiber in range(frame.flux.shape[0]) :
+
+                    # estimate and subtract continuum for this fiber specifically
+
+                    cont = np.interp(skymodel.wave,skymodel.wave[in_cont],skymodel.flux[fiber][in_cont])
+                    skylines = skymodel.flux[fiber] - cont
+                    skylines[skylines<threshold] = 0
```

Comment:
I was originally imagining having a sky line template rather than figuring it out on the fly.  One feature there is that one could imagine handling the wings of the line a little more gracefully than setting them to zero.

Reply:
part of code removed